zebra-network/src/address_book.rs

```diff
@@ -266,6 +266,19 @@ impl AddressBook {
         self.local_listener
     }
 
+    /// Get the active addresses in `self` in random order with sanitized timestamps,
+    /// including our local listener address.
+    ///
+    /// Limited to a the number of peer addresses Zebra should give out per `GetAddr` request.
+    pub fn fresh_get_addr_response(&self) -> Vec<MetaAddr> {
+        let now = Utc::now();
+        let mut peers = self.sanitized(now);
+        let address_limit = peers.len().div_ceil(ADDR_RESPONSE_LIMIT_DENOMINATOR);
+        peers.truncate(MAX_ADDRS_IN_MESSAGE.min(address_limit));
+
+        peers
+    }
+
     /// Get the active addresses in `self` in random order with sanitized timestamps,
     /// including our local listener address.
     pub fn sanitized(&self, now: chrono::DateTime<Utc>) -> Vec<MetaAddr> {
```

Comment:
Let's see if this works, feel free to revert if it doesn't.

```suggestion
    pub(crate) fn sanitized(&self, now: chrono::DateTime<Utc>) -> Vec<MetaAddr> {
```